packages/web/src/components/Modal.js

```diff
@@ -0,0 +1,74 @@
+// Module imports
+import {
+	useCallback,
+	useEffect,
+} from 'react'
+import { createPortal } from 'react-dom'
+import PropTypes from 'prop-types'
+
+
+
+
+
+// Local imports
+import { Button } from 'components/Button'
+import { useModals } from 'contexts/ModalsContext'
+
+
+
+
+
+function Modal(props) {
+	const {
+		children,
+		name,
+		title,
+	} = props
+	const { closeModal } = useModals()
+
+	const close = useCallback(() => closeModal(name), [
+		closeModal,
+		name,
+	])
+
+	useEffect(() => {
+		const handleEscape = event => {
+			if (event.code.toLowerCase() === 'escape') {
```

Comment:
This code property is not always defined. For example, clicking the auto-completed value for the field causes a JS error:

![image](https://user-images.githubusercontent.com/26912533/111095708-05e61a80-850c-11eb-91be-9ffd3e47cfae.png)

Reply:
Oooooooh, interesting... I figured mouse buttons would still present a `code`, but I guess not. TIL. This is an easy fix, though.